src/Agent/NewRelic/Agent/Core/Metrics/MetricNameService.cs

```diff
@@ -129,27 +121,38 @@ private static string StripParameters(string url)
             return url;
         }
 
-        private static string RenameUsingRegexRules(string input, IEnumerable<RegexRule> rules)
+        private static bool TryRenameUsingRegexRules(string input, IEnumerable<RegexRule> rules, out string result)
         {
+            result = null;
```

Comment:
Initializing `result` to null will cause this method to return null if no rules match (or if there are no rules in the `IEnumerable`).  This is causing a fair number of unit test failures.  I tested initializing `result` to `input` and all the unit tests passed.  I tried committing this change to a new branch on your fork, but I got a permissions error when I tried pushing it up, so I'll have to continue being a "backseat coder" and ask you to make this change yourself.

Reply:
I'm not sure why you were unable to commit.  I can check into it...  Can you send along the specific error message?

As for the error you mentioned... for some reason when I ran my tests after making the initial changes they succeeded even though, as you note, they shouldn't have.  Later, they started failing for the reason you mentioned above. After setting result = input they continued to fail.  I'm going to do a clean -dfx and give it another go to see if I can clear things out.